### PR TITLE
Follow-up work to #9102

### PR DESCRIPTION
**Pull request by ben-albrecht**
Follow-up PR to #9102:

- Use `python -S` flag to avoid any `sys.path` prepending (looking at you Ubuntu)
- Add a helpful message that suggests setting `CHPL_PIP` if `curl` failed (likely due to an outdated openssl)
  - Maybe we should also suggest updating openssl?
- Stop using `$(PYTHON)` to call `python`. This variable was intended to check the python version, not pin the binary we should be using.

### CODE REVIEW

File: third-party/chpl-venv/Makefile [hunk 1]
```diff
@@ -30,16 +30,20 @@ $(CHPL_VENV_INSTALL_DIR):
...
 	  export PYTHONUSERBASE=$(CHPL_VENV_INSTALL_DIR) && \
-	  $(PYTHON) get-pip.py --user) \
+	  python get-pip.py --user) \
 	fi
 
 # Install virtualenv program.
 $(CHPL_VENV_VIRTUALENV): check-exes $(PIP)
 	export PYTHONPATH="$(PIPLIBS):$$PYTHONPATH" && \
-	$(PYTHON) $(PIP) install -U --force-reinstall --ignore-installed \
+	python -S $(PIP) install -U --force-reinstall --ignore-installed \
```
Comment (ronawho) [inline third-party/chpl-venv/Makefile]: Can we unify on `$(PYTHON)` vs `python`?

**Comment by ben-albrecht** [comment]
Spot checked on the Ubuntu / CentOS systems that required the `-S` flag, as well as OS X, SLES 12, and CLE 6.

File: third-party/chpl-venv/Makefile [hunk 2]
```diff
@@ -30,16 +30,20 @@ $(CHPL_VENV_INSTALL_DIR):
...
 	 --prefix=$(CHPL_VENV_INSTALL_DIR) $(CHPL_PIP_INSTALL_PARAMS) $(shell cat virtualenv.txt)
```
[the remark this answers is not in the document]
Comment (ben-albrecht) [reply third-party/chpl-venv/Makefile]: 👍 See https://github.com/chapel-lang/chapel/pull/9330

**Comment by awallace-cray** [comment]
2018-04-25 
git checkout release/1.17
git cherry-pick 7b1040447aaaedd7d3d44fac30f963e14a9209fc 853f64e77717c2eb4473085d7dbc4c3f96c24520

**Pull request merged**